### PR TITLE
GH-1029: Refactor entity-loader to make it handle multiple items

### PR DESCRIPTION
**Pull request by dunqan**

### CODE REVIEW

File: projects/core/src/state/utils/loader/loader.action.ts [hunk 1]
```diff
@@ -5,10 +5,11 @@ export const LOADER_FAIL_ACTION = '[LOADER] FAIL';
 export const LOADER_SUCCESS_ACTION = '[LOADER] SUCCESS';
 
 export interface LoaderMeta {
+  entityType: string;
   loader: {
-    type?: string;
     load?: boolean;
     error?: any;
+    success?: any;
```
Comment (WeizhengSap) [inline projects/core/src/state/utils/loader/loader.action.ts]: success is any or boolean?

File: projects/core/src/cms/store/selectors/component.selectors.ts [hunk 2]
```diff
@@ -21,17 +27,20 @@ export const getComponentEntities: MemoizedSelector<
   getComponentEntitiesSelector
 );
 
+export const componentStateSelectorFactory = (
+  uid
```
Comment (WeizhengSap) [inline projects/core/src/cms/store/selectors/component.selectors.ts]: add type to 'uid'

File: projects/core/src/cms/store/selectors/component.selectors.ts [hunk 3]
```diff
@@ -21,17 +27,20 @@ export const getComponentEntities: MemoizedSelector<
...
+): MemoizedSelector<StateWithCms, LoaderState<any>> => {
+  return createSelector(
+    getComponentState,
+    entities => entityStateSelector(entities, uid)
+  );
+};
+
 export const componentSelectorFactory = (
   uid
```
Comment (WeizhengSap) [inline projects/core/src/cms/store/selectors/component.selectors.ts]: add type to 'uid'

File: projects/core/src/state/utils/entity/entity.reducer.ts [hunk 4]
```diff
@@ -17,19 +17,37 @@ export function entityReducer<T>(
     state: EntityState<T> = initialEntityState,
     action: EntityAction
   ): EntityState<T> => {
+    let ids;
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/state/utils/entity/entity.reducer.ts]: 👍 thanks

**Pull request merged**